ledger/accountdb.go

```diff
@@ -136,60 +148,95 @@ func writeCatchpointStagingCreatable(ctx context.Context, tx *sql.Tx, addr basic
 	return nil
 }
 
-func writeCatchpointStagingBalances(ctx context.Context, tx *sql.Tx, bals []encodedBalanceRecord) error {
-	insertStmt, err := tx.PrepareContext(ctx, "INSERT INTO catchpointbalances(address, data) VALUES(?, ?)")
+func writeCatchpointStagingBalances(ctx context.Context, tx *sql.Tx, bals []encodedBalanceRecord, proto config.ConsensusParams) error {
```

Comment:
I'm pretty confident that this isn't what you want to do:
Using the genesis config.ConsensusParams here means that accounts that would be using fast catchup would calculate the normalized online balances based on the genesis params and not by the most recent params.

I would suggest you'll:
1. add the calculated normalized online balances to the catchpoint file
2. "restore" it in the same way, without recalculating anything.
3. add the calculated normalized online balances to the hashing of the account, so that the merkletrie would include that.
4. update the version of the catchpoint file, so that it would "break" compatibility with older files.

I have yet had the need to perform #4, so I don't know how hard it would be. ( shouldn't be too hard ).

I will be available to help you with this one if you'd like next week.

Reply:
The `proto` is used only for the value of `proto.RewardUnit`, which cannot change in a given network, so it's OK to pass the proto for the genesis block or the proto for a recent block.

Reply:
In that case, we should always use the genesis round proto in the call to 
`err = accountsNewRound(tx, deltas[i], creatableDeltas[i], protos[i+1])` as well.

Reply:
That seems like a reasonable idea; thanks!  Pushed.